flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliTableauResultView.java

```diff
@@ -126,34 +122,29 @@ private void checkAndCleanUpQuery(boolean cleanUpQuery) {
     private void printBatchResults(AtomicInteger receivedRowCount) {
         final List<RowData> resultRows = waitBatchResults();
         receivedRowCount.addAndGet(resultRows.size());
-        PrintUtils.printAsTableauForm(
-                resultDescriptor.getResultSchema(),
-                resultRows.iterator(),
-                terminal.writer(),
-                sessionTimeZone);
+        TableauStyle style =
+                PrintStyle.tableauWithDataInferredColumnWidths(
+                        resultDescriptor.getResultSchema(),
+                        resultDescriptor.getRowDataStringConverter(),
+                        PrintStyle.DEFAULT_MAX_COLUMN_WIDTH,
+                        false,
+                        false);
+        style.print(resultRows.iterator(), terminal.writer());
     }
 
     private void printStreamingResults(AtomicInteger receivedRowCount) {
-        List<Column> columns = resultDescriptor.getResultSchema().getColumns();
-        final String[] fieldNames =
-                Stream.concat(
-                                Stream.of(PrintUtils.ROW_KIND_COLUMN),
-                                columns.stream().map(Column::getName))
-                        .toArray(String[]::new);
-
-        final int[] colWidths =
-                PrintUtils.columnWidthsByType(
-                        columns,
+        TableauStyle style =
+                PrintStyle.tableauWithTypeInferredColumnWidths(
+                        resultDescriptor.getResultSchema(),
+                        resultDescriptor.getRowDataStringConverter(),
```

Comment:
My understanding from last time we talked about this is that we can't easily convert the CLI classes to handle directly `String`s instead of `RowData`, because we perform the materialization here.

Reply:
No, currently the materialization happens on the server side. Given the current code base I would stick to me comment above. After future refactorings this view might change, because materialization might happen on the client side.

Reply:
I'm fine if you ignore my comment. But the architecture will need some rework soon.

Reply:
I would personally prefer to ignore this comment now, since this change is already pretty big and I feel modifying the way views works is even a bigger change.

I'm ok with reworking the architecture, but then we need to make sure everything passing through `Executor` needs to be some kind of primitive type, that is as you sad we probably need to rework `executeOperation` as well, as the other types.